app/api/tests/test_api.py

```diff
@@ -1199,6 +1200,58 @@ def test_can_filter_doc_without_anotation(self):
     def doCleanups(cls):
         remove_all_role_mappings()
 
+class TestImportExportIntegrity(APITestCase):
+    """Tests that check for equality between imported and exported data of a file. """
+    @classmethod
+    def setUpTestData(cls):
+        cls.super_user_name = 'super_user_name'
+        cls.super_user_pass = 'super_user_pass'
+        create_default_roles()
+        super_user = User.objects.create_superuser(username=cls.super_user_name,
+                                                   password=cls.super_user_pass,
+                                                   email='fizz@buzz.com')
+        cls.classification_project = mommy.make('TextClassificationProject',
+                                                users=[super_user], project_type=DOCUMENT_CLASSIFICATION)
+
+        cls.classification_upload_url = reverse(viewname='doc_uploader', args=[cls.classification_project.id])
+        cls.classification_download_url = reverse(viewname='doc_downloader', args=[cls.classification_project.id])
+        assign_user_to_role(project_member=super_user, project=cls.classification_project,
+                            role_name=settings.ROLE_PROJECT_ADMIN)
+    def setUp(self):
+        self.client.login(username=self.super_user_name,
+                          password=self.super_user_pass)
+
+    def load_test_helper(self, upload_url, download_url, project_id, filename, format, **kwargs):
+        import_data = []
+        parser = TextUploadAPI.select_parser(format)
+
+        with open(os.path.join(DATA_DIR, filename), 'rb') as f:
+            self.client.post(upload_url, data={'file': f, 'format': format})
+            f.seek(0)
+            imported = parser.parse(f)
+            import_data = next(imported)
+
+        export_data = self.client.get(download_url, data={'q': format}).data
+        print(export_data)
+        self.assertTrue(len(import_data) == len(export_data), 'Integritycheck failed. Lengths differ.')
+
+        for im, up in zip(import_data, export_data):
+            self.assertTrue(list(im.items())[0] == list(up.items())[1], 'Integritycheck failed. Items differ.')
```

Comment:
why `[0]` and `[1]`? please explain what exactly is checked here

Reply:
As import and export data are structured differently, key-value-pairs are stored at different locations in import and export data.
Furthermore since this a prototype I've left print-Statements whose output one can find in the first failed check. In this you will also see that one of the tests failed. You can find an explanation on why it failed on skype including a question on how to handle it.

Reply:
indexing with numbers is not the best method for dicts. better to use `im['text'] == up['text']`